app/views/katello/sync_management/index.html.erb

```diff
@@ -1,17 +1,19 @@
-<%= javascript 'katello/sync_management' %>
+<% content_for(:javascripts) do -%>
+  <script src="/assets/katello/common/index.js"></script>
+  <script src="/assets/katello/sync_management.js"></script>
```

Comment:
```suggestion
  <%= javascript 'katello/common/index' %>
  <%= javascript 'katello/sync_management' %>
```
this makes things load the right files but in the wrong order:

```
<script type="text/javascript">
      const html = "  \n  \n  <script>\n    localize({\n      \"cancel\": \"Cancel\",\n      \"cancelled\": \"Sync Canceled\",\n      \"error\": \"Error\",\n      \"complete\": \"Sync complete.\",\n      \"no_start_time\": \"No start time currently available.\"\n    });\n\n      KT.repo_status = \$.parseJSON(\'{}\');\n      KT.permissions = { \"syncable\" : false };\n  <\/script>\n  <script src=\"/assets/katello/common/vendor-9f3c77c20de190d174986979ad7d0d5cee9362a9a280adabf643e57ee4259155.js\"><\/script>\n  <script src=\"/assets/katello/common-41006bb366e1429ef7397de66336e76035be453bc9bac600cd845b9571266232.js\"><\/script>\n  <script src=\"/assets/katello/common/index-41006bb366e1429ef7397de66336e76035be453bc9bac600cd845b9571266232.js\"><\/script><script src=\"/assets/katello/sync_management-db168612191a6dc850c8d5cd28375ee6af25d8fd011e62638f11f7fd903f5025.js\"><\/script>\n"
      load_dynamic_javascripts(html);
    </script>
```

and well, this gives the localize error again

Reply:
```suggestion
  <%= javascript_include_tag 'katello/common/index' %>
  <%= javascript_include_tag 'katello/sync_management' %>
```

This works.
Now, wth is the difference between `javascript` and `javascript_include_tag`?!

Reply:
Ah, it's already wrapping it in `content_for`… Weird

https://github.com/theforeman/foreman/blob/262b5caceab9c3f71d0b80f189cc724389593fb0/app/helpers/layout_helper.rb#L98-L100

Reply:
Thinking more, this makes sense!

* Your original intent was "have the script load tags and the following embedded code in one block, so that it gets executed in the right order", but had the bug that it ignored asset pipeline paths.
* Using `javascript` creates a `content_for(:javascripts…)` block, which gets yielded into the `:javascripts` section individually (so the `=` of `<%=` didn't really do anything) and thus the order is fucked up
* Using `javascript_include_tag` creates script tags that honor the asset pipeline *AND* prints them directly where the code is, thus resulting in the correct order as one block, which then gets yielded into `:javascripts`.

Reply:
```
>>> KT
<<< Object { utils: $(n), common: {…}, widget: {}, getData: getData(n), orgswitcher: {…}, menu: {…}, object: {…} }

>>> load_dynamic_javascripts("<script src=\"/assets/katello/sync_management-db168612191a6dc850c8d5cd28375ee6af25d8fd011e62638f11f7fd903f5025.js\"><\/script>")
<<< undefined

>>> KT
<<< Object { utils: $(n), common: {…}, widget: {}, getData: getData(n), orgswitcher: {…}, menu: {…}, object: {…}, content: {…}, content_actions: {…} }
```

So reloading that sync management script works. WTH.

Reply:
Could it be something re-initializes `KT`, thus overwriting what this code did?